src/components/post/PostView.jsx

```diff
@@ -55,14 +55,27 @@ class PostView extends Component {
   }
 
   handlePostUpdate = (post) => {
-    this.setState((prevState) => ({
-      // no longer editing the post
-      editingPostId: null,
-      // update post that got edited
-      posts: prevState.posts.map(
-        (p) => (p.id === post.id ? Object.assign(p, post) : p),
-      ),
-    }));
+    postService.updateUserPosts(post).then(() => {
```

Comment:
`updateUserPosts` can return an error but we never caught any errors here. So we need to put a `catch` in both this function and in the `handleDelete` function below